packages/core/contracts/cross-chain-oracle/GovernorHub.sol

```diff
@@ -0,0 +1,48 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "../common/implementation/Lockable.sol";
+import "./ParentMessengerInterface.sol";
+
+/**
+ * @title Governance relayer contract to be deployed on Ethereum that receives messages from the owner (Governor) and
+ * sends them to spoke contracts on child chains.
+ */
+contract GovernorHub is Ownable, Lockable {
+    // Associates chain ID with ParentMessenger contract to use to send governance actions to that chain's GovernorSpoke
+    // contract.
+    mapping(uint256 => ParentMessengerInterface) public messengers;
+
+    event RelayedGovernanceRequest(uint256 indexed chainId, address indexed messenger, address indexed to, bytes data);
+    event SetParentMessenger(uint256 indexed chainId, address indexed parentMessenger);
+
+    /**
+     * @notice Set new ParentMessenger contract for chainId.
+     * @param chainId child network that messenger contract will communicate with.
+     * @param messenger ParentMessenger contract that sends messages to ChildMessenger on network with ID `chainId`.
+     * @dev Only callable by the owner (presumably the Ethereum Governor contract).
+     */
+    function setMessenger(uint256 chainId, ParentMessengerInterface messenger) public nonReentrant() onlyOwner {
+        require(address(messenger) != address(0), "Invalid messenger contract");
+        messengers[chainId] = messenger;
+        emit SetParentMessenger(chainId, address(messenger));
+    }
+
+    /**
+     * @notice This should be called in order to relay a governance request to the `GovernorSpoke` contract
+     * deployed to the child chain associated with `chainId`.
+     * @param chainId network that messenger contract will communicate with
+     * @param to Contract on child chain to send message to
+     * @param data Message to send
+     * @dev Only callable by the owner (presumably the Ethereum Governor contract).
+     */
+    function relayGovernance(
+        uint256 chainId,
+        address to,
+        bytes memory data
+    ) external nonReentrant() onlyOwner {
+        messengers[chainId].sendMessageToChild(abi.encode(to, data));
```

Comment:
is there any point in checking that the `chainId` contains a `messenger`? I guess this will revert if that does not exist due to calling a function on an address that does not implement that interface?

Reply:
Yeah, and only the `owner` can set the address so we can assume the `messenger` is not a malicious contract?

Reply:
It's less about it being malicious, more about erroring in a readable way. but I agree with you though that it's somewhat redundant given the calling context.